pkg/controller/direct/registry/registry.go

```diff
@@ -32,6 +33,36 @@ type registry struct {
 	registrations map[schema.GroupKind]*registration
 }
 
+// staticRegistrations contain all the GKs and their highest versions supported
+// via direct controllers.
+// It's the static set of the direct controllers, and it should contain all the
+// GKs NOT based on TF, DCL, or handwritten IAM types.
+var staticRegistrations = map[schema.GroupKind]string{
+	schema.GroupKind{Group: "logging.cnrm.cloud.google.com", Kind: "LoggingLogMetric"}:                                       k8s.KCCAPIVersionV1Beta1,
+	schema.GroupKind{Group: "cloudbuild.cnrm.cloud.google.com", Kind: "CloudBuildWorkerPool"}:                                k8s.KCCAPIVersionV1Beta1,
+	schema.GroupKind{Group: "apikeys.cnrm.cloud.google.com", Kind: "APIKeysKey"}:                                             k8s.KCCAPIVersionV1Beta1,
+	schema.GroupKind{Group: "monitoring.cnrm.cloud.google.com", Kind: "MonitoringDashboard"}:                                 k8s.KCCAPIVersionV1Beta1,
+	schema.GroupKind{Group: "dataform.cnrm.cloud.google.com", Kind: "DataformRepository"}:                                    k8s.KCCAPIVersionV1Alpha1,
+	schema.GroupKind{Group: "alloydb.cnrm.cloud.google.com", Kind: "AlloyDBCluster"}:                                         k8s.KCCAPIVersionV1Beta1,
+	schema.GroupKind{Group: "dataflow.cnrm.cloud.google.com", Kind: "DataflowFlexTemplateJob"}:                               k8s.KCCAPIVersionV1Beta1,
+	schema.GroupKind{Group: "gkehub.cnrm.cloud.google.com", Kind: "GKEHubFeatureMembership"}:                                 k8s.KCCAPIVersionV1Beta1,
+	schema.GroupKind{Group: "networkconnectivity.cnrm.cloud.google.com", Kind: "NetworkConnectivityServiceConnectionPolicy"}: k8s.KCCAPIVersionV1Alpha1,
+	schema.GroupKind{Group: "redis.cnrm.cloud.google.com", Kind: "RedisCluster"}:                                             k8s.KCCAPIVersionV1Alpha1,
+	schema.GroupKind{Group: "tags.cnrm.cloud.google.com", Kind: "TagsTagKey"}:                                                k8s.KCCAPIVersionV1Beta1,
+	schema.GroupKind{Group: "sql.cnrm.cloud.google.com", Kind: "SQLInstance"}:                                                k8s.KCCAPIVersionV1Beta1,
+	schema.GroupKind{Group: "compute.cnrm.cloud.google.com", Kind: "ComputeForwardingRule"}:                                  k8s.KCCAPIVersionV1Beta1,
+	schema.GroupKind{Group: "privateca.cnrm.cloud.google.com", Kind: "PrivateCACAPool"}:                                      k8s.KCCAPIVersionV1Beta1,
+	schema.GroupKind{Group: "securesourcemanager.cnrm.cloud.google.com", Kind: "SecureSourceManagerInstance"}:                k8s.KCCAPIVersionV1Alpha1,
+}
+
+func GetDirectGVKs() []schema.GroupVersionKind {
```

Comment:
Allow-list may not be a good idea because it is hard to keep track and it buries into the code. To encourage external contributors, we shall be careful every additional step we ask them to do.

If you want to list all the direct resource and make that self-maintained, you can iterate the CRDs and call  
 `IsDirectByGK`.

Reply:
This is not doable for unit tests/local call of functions () that doesn't dynamically register the direct controllers. E.g. when I ran `make resource-docs`, the singleton registry is empty so there is no direct GK for the caller to identify. That's why I think it's important to also have a static registry.

We can also just go through all the CRDs whenever we want to figure out what direct CRDs are.

Reply:
I think there's some confusion about the direct controller registry: when a direct controller is registered, it does not mean it would be used. It is the CRD's absence of tf2crd or dcl2crd label that determines whether a resource is direct resource or not. 

So the label could be frequently changed and this allow list approach could be error-prone (and out of date easily).

Reply:
As discussed, I replaced the static allowlist with a CRD go-through.